<!DOCTYPE html>
<html lang="zh-CN">

<head>
    <meta charset="UTF-8">
    <meta name="viewport"
        content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0,minimal-ui:ios">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title></title>
    <link rel="stylesheet" href="">

</head>

<body>
    <script>
        /**
    当接收到一个代表错误的 HTTP 状态码时，从 fetch()返回的 Promise 不会被标记为 reject， 即使该 HTTP 响应的状态码是 404 或 500。
    相反，它会将 Promise 状态标记为 resolve （但是会将 resolve 的返回值的 ok 属性设置为 false ），
    仅当网络故障时或请求被阻止时，才会标记为 reject。
    默认情况下，fetch 不会从服务端发送或接收任何 cookies, 如果站点依赖于用户 session，则会导致未经认证的请求（要发送 cookies，
    必须设置 credentials 选项）。自从2017年8月25日后，默认的credentials政策变更为same-originFirefox也在61.0b13中改变默认值
        */
        var url = 'https://www.easy-mock.com/mock/5b3981c7c4dcd03fb7171981/example/medicineSearch';
        var data = { username: 'example' };
        fetch(url, {
            method: 'POST', // or 'PUT'
            body: JSON.stringify(data), // data can be `string` or {object}!
            headers: new Headers({
                'Content-Type': 'application/json'
            })
        }).then(res => {
            if (res.status == 200) {
                return res.json()//json是返回的response提供的一个方法,会把返回的json字符串反序列化成对象,也被包装成一个Promise了
            } else {
                return res
            }
            console.log(res)
        })//
            .catch(error => console.error('Error:', error))
            .then(response => console.log('Success:', response));// 如果状态吗是404 500 会是resolve 而不是 reject
    </script>
</body>

</html>